### ISSUE TITLE
bug: Exception java.lang.IllegalStateException: Either grant() or deny() has been already called

### ISSUE THREAD

**Issue by ecc521** [ionic-team/capacitor#6879]
## Bug Report

App crash on Android. Previously observed in #6094, and still exists. I've attached two sample traces. This issue occurs with low frequency, though there are hits online as for what might be causing it. 

Trace 1:
```
Exception java.lang.IllegalStateException: Either grant() or deny() has been already called.
  at WV.bC0.grant (chromium-TrichromeWebViewGoogle.aab-stable-579016631:97)
  at com.getcapacitor.BridgeWebChromeClient.lambda$onPermissionRequest$2 (BridgeWebChromeClient.java:122)
  at com.getcapacitor.BridgeWebChromeClient$$ExternalSyntheticLambda6.onPermissionSelect
  at com.getcapacitor.BridgeWebChromeClient.lambda$new$0 (BridgeWebChromeClient.java:65)
  at com.getcapacitor.BridgeWebChromeClient.$r8$lambda$xze8TXtu7pSZVmwGgEN_UZU2Gfg
  at com.getcapacitor.BridgeWebChromeClient$$ExternalSyntheticLambda4.onActivityResult
  at androidx.activity.result.ActivityResultRegistry$1.onStateChanged (ActivityResultRegistry.java:149)
  at androidx.lifecycle.LifecycleRegistry$ObserverWithState.dispatchEvent (LifecycleRegistry.kt:314)
  at androidx.lifecycle.LifecycleRegistry.forwardPass (LifecycleRegistry.kt:251)
  at androidx.lifecycle.LifecycleRegistry.sync (LifecycleRegistry.kt:287)
  at androidx.lifecycle.LifecycleRegistry.moveToState (LifecycleRegistry.kt:136)
  at androidx.lifecycle.LifecycleRegistry.handleLifecycleEvent (LifecycleRegistry.kt:119)
  at androidx.lifecycle.ReportFragment$Companion.dispatch$lifecycle_runtime_release (ReportFragment.kt:192)
  at androidx.lifecycle.ReportFragment$LifecycleCallbacks.onActivityPostStarted (ReportFragment.kt:121)
  at android.app.Activity.dispatchActivityPostStarted (Activity.java:1496)
  at android.app.Activity.performStart (Activity.java:8611)
  at android.app.ActivityThread.handleStartActivity (ActivityThread.java:3922)
  at android.app.servertransaction.TransactionExecutor.performLifecycleSequence (TransactionExecutor.java:221)
  at android.app.servertransaction.TransactionExecutor.cycleToPath (TransactionExecutor.java:201)
  at android.app.servertransaction.TransactionExecutor.executeLifecycleState (TransactionExecutor.java:173)
  at android.app.servertransaction.TransactionExecutor.execute (TransactionExecutor.java:97)
  at android.app.ActivityThread$H.handleMessage (ActivityThread.java:2437)
  at android.os.Handler.dispatchMessage (Handler.java:106)
  at android.os.Looper.loopOnce (Looper.java:211)
  at android.os.Looper.loop (Looper.java:300)
  at android.app.ActivityThread.main (ActivityThread.java:8294)
  at java.lang.reflect.Method.invoke
  at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run (RuntimeInit.java:580)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:1028)
```

Trace 2:

```
Exception java.lang.IllegalStateException: Either grant() or deny() has been already called.
  at WV.bC0.grant (chromium-TrichromeWebViewGoogle.aab-stable-579013831:97)
  at com.getcapacitor.BridgeWebChromeClient.lambda$onPermissionRequest$2 (BridgeWebChromeClient.java:122)
  at com.getcapacitor.BridgeWebChromeClient$$ExternalSyntheticLambda6.onPermissionSelect
  at com.getcapacitor.BridgeWebChromeClient.lambda$new$0 (BridgeWebChromeClient.java:65)
  at com.getcapacitor.BridgeWebChromeClient.$r8$lambda$xze8TXtu7pSZVmwGgEN_UZU2Gfg
  at com.getcapacitor.BridgeWebChromeClient$$ExternalSyntheticLambda4.onActivityResult
  at androidx.activity.result.ActivityResultRegistry$1.onStateChanged (ActivityResultRegistry.java:149)
  at androidx.lifecycle.LifecycleRegistry$ObserverWithState.dispatchEvent (LifecycleRegistry.kt:314)
  at androidx.lifecycle.LifecycleRegistry.forwardPass (LifecycleRegistry.kt:251)
  at androidx.lifecycle.LifecycleRegistry.sync (LifecycleRegistry.kt:287)
  at androidx.lifecycle.LifecycleRegistry.moveToState (LifecycleRegistry.kt:136)
  at androidx.lifecycle.LifecycleRegistry.handleLifecycleEvent (LifecycleRegistry.kt:119)
  at androidx.lifecycle.ReportFragment$Companion.dispatch$lifecycle_runtime_release (ReportFragment.kt:192)
  at androidx.lifecycle.ReportFragment$LifecycleCallbacks.onActivityPostStarted (ReportFragment.kt:121)
  at android.app.Activity.dispatchActivityPostStarted (Activity.java:1420)
  at android.app.Activity.performStart (Activity.java:8352)
  at android.app.ActivityThread.handleStartActivity (ActivityThread.java:4060)
  at android.app.servertransaction.TransactionExecutor.performLifecycleSequence (TransactionExecutor.java:221)
  at android.app.servertransaction.TransactionExecutor.cycleToPath (TransactionExecutor.java:201)
  at android.app.servertransaction.TransactionExecutor.cycleToPath (TransactionExecutor.java:183)
  at android.app.servertransaction.TransactionExecutor.executeCallbacks (TransactionExecutor.java:132)
  at android.app.servertransaction.TransactionExecutor.execute (TransactionExecutor.java:95)
  at android.app.ActivityThread$H.handleMessage (ActivityThread.java:2438)
  at android.os.Handler.dispatchMessage (Handler.java:106)
  at android.os.Looper.loopOnce (Looper.java:226)
  at android.os.Looper.loop (Looper.java:313)
  at android.app.ActivityThread.main (ActivityThread.java:8663)
  at java.lang.reflect.Method.invoke
  at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run (RuntimeInit.java:571)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:1135)
```

### Capacitor Version

```
  @capacitor/cli: 5.2.2
  @capacitor/core: 5.2.2
  @capacitor/android: 5.2.2
  @capacitor/ios: 5.2.2
```

### Platform(s)
Android

### Current Behavior
App crash. 

### Expected Behavior
No crash. 

### Other Technical Details

`npm --version` output: 9.8.1

`node --version` output: v20.0.0

### Additional Context

The app is requesting both camera and microphone permissions using navigator.getUserMedia

**Comment by ecc521** [comment]
Alright, I don't believe I'm going to be able to give a reliably reproducing sample app for this bug, but what I can tell you is that BridgeWebChromeClient.java is the only file that grants or denies permissions, and that somehow two calls are being made on one permission request. 

I suspect the most likely case here is that two permission requests are being triggered, and that is somehow causing permissionListener to fire twice. I'm not the best with Java, but I'm pretty sure that is the problem. 

Basically, somewhere [in this code](https://github.com/ionic-team/capacitor/blob/157d02fedb364d1eaf7541662f22774809c77723/android/capacitor/src/main/java/com/getcapacitor/BridgeWebChromeClient.java#L117C0-L130C3) grant or deny is being called twice on a permission request.

**Comment by jcesarmobile** [comment]
BridgeWebChromeClient request the permissions when your app is using some code that requires permissions, such as `navigator.geolocation` or `mediadevices.getusermedia`, so if you are not able to provide the code that request the permissions, we are not able to help you, sorry.
